sql/analyzer/rules.go

```diff
@@ -531,14 +534,82 @@ func resolveNaturalJoins(ctx *sql.Context, a *Analyzer, n sql.Node) (sql.Node, e
 
 		projections := append(append(common, left...), right...)
 
-		return plan.NewProject(
+		transformed = plan.NewProject(
 			projections,
 			plan.NewInnerJoin(
 				join.Left,
 				join.Right,
 				expression.JoinAnd(conditions...),
 			),
-		), nil
+		)
+
+		return transformed, nil
+	})
+
+	if err != nil || transformed == nil {
```

Comment:
can `transformed` be nil in any other case than `err` being `!= nil`?

Reply:
Yes, either there isn't a natural join to transform or it's transformed intto a `cross join` instead of a `inner join`, transformed stays as nil